UI/src/view/linear_asset/laneModellingStyle.js

```diff
@@ -32,18 +32,33 @@
       }
 
       var linearAssetsWithType = _.flatten(_.map(relevantLinears, function(linearAsset) {
-        var hasAsset = me.hasValue(linearAsset);
+        var hasAsset = me.hasValue(linearAsset) || !_.isUndefined(linearAsset.isViewOnly);
         var type = {type: 'line'};
```

Comment:
Since you use hasAsset and type in at the same moments then you could replace this by var hasAssetAndType = {hasAsset: hasAsset, type: 'line'} and replace the instance of those hasAsset and type with only this one

Reply:
Done, merged three suggestions of this nature